packages/manager/modules/pci/src/projects/project/private-registry/list/list.controller.js

```diff
@@ -0,0 +1,91 @@
+export default class {
+  /* @ngInject */
+  constructor(
+    $state,
+    $stateParams,
+    CucCloudMessage,
+    CucControllerHelper,
+    CucServiceHelper,
+    privateRegistryService,
+    PRIVATE_REGISTRY_STATUS_MAP,
+    PRIVATE_REGISTRY_STATUS,
+  ) {
+    this.$state = $state;
+    this.$stateParams = $stateParams;
+    this.projectId = $stateParams.projectId;
+    this.CucCloudMessage = CucCloudMessage;
+    this.cucControllerHelper = CucControllerHelper;
+    this.cucServiceHelper = CucServiceHelper;
+    this.privateRegistryService = privateRegistryService;
+    this.PRIVATE_REGISTRY_STATUS_MAP = PRIVATE_REGISTRY_STATUS_MAP;
+    this.PRIVATE_REGISTRY_STATUS = PRIVATE_REGISTRY_STATUS;
+  }
+
+  $onInit() {
+    this.loadMessages();
+    this.getRegistryList(false);
+  }
+
+  loadMessages() {
+    this.CucCloudMessage.unSubscribe('pci.projects.project.private-registry.list');
+    this.messageHandler = this.CucCloudMessage.subscribe('pci.projects.project.private-registry.list', { onMessage: () => this.refreshMessages() });
+  }
+
+  refreshMessages() {
+    this.messages = this.messageHandler.getMessages();
+  }
+
+  getRegistryList(clearCache) {
+    this.registryList = this.cucControllerHelper.request.getArrayLoader({
+      loaderFunction: () => this.privateRegistryService.getRegistryList(this.projectId, clearCache)
+        .catch(error => this.cucServiceHelper.errorHandler('private_registry_query_error')(error)),
+    });
+    return this.registryList.load();
+  }
+
+  createRegistry() {
+    return this.$state.go('pci.projects.project.private-registry.list.create', { fromState: 'list' });
```

Comment:
It should be in state `resolves` too : https://github.com/ovh-ux/manager/blob/master/packages/manager/modules/pci/src/projects/project/instances/instances.routing.js#L19

Reply:
I can put it in state resolves but the origin state can be different so i am sending the `fromState` as a parameter here. The other way of doing is to check `$transition` and get the previous state.